tools/curl/gurl/__init__.py

```diff
@@ -67,43 +71,44 @@ def parse_binary(raw_request, raw_response):
     }
 
 
+def _bytes_reduce(a:bytearray, b:bytes):
+    a.extend(b)
+    return a
+
+
+def _extract_req(meta_req_res):
+    raw_data = map(lambda b: b.data, meta_req_res.req)
+    return reduce(_bytes_reduce, raw_data, bytearray())
+
+
+def _extract_res(meta_req_res):
+    raw_data = map(lambda b: b.data, meta_req_res.res)
```

Comment:
Same as before, use https://docs.python.org/3/library/operator.html#operator.attrgetter

Reply:
Ok, I'll use this to provide better names also.